Data/models/DataController.swift

```diff
@@ -30,7 +30,7 @@ private let log = Logger.browserLogger
 // Follow the stack design from http://floriankugler.com/2013/04/02/the-concurrent-core-data-stack/
 
 public class DataController: NSObject {
-    public static let shared = DataController()
+    public static var shared: DataController? = DataController()
```

Comment:
What's the reason for making this an optional? Even if we allow it to be reset, we don't need to make it an optional. Just not a constant.